File: src/main/scala/org/openapitools/client/api/CodingList.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import CodingList._

case class CodingList (
  metadata: Option[Metadata],
pagination: Option[Pagination],
data: Option[List[Coding]])

object CodingList {
  import DateTimeCodecs._

  implicit val CodingListCodecJson: CodecJson[CodingList] = CodecJson.derive[CodingList]
  implicit val CodingListDecoder: EntityDecoder[CodingList] = jsonOf[CodingList]
  implicit val CodingListEncoder: EntityEncoder[CodingList] = jsonEncoderOf[CodingList]
}

File: API/Client/CodingList.js
goog.provide('API.Client.CodingList');

/**
 * @record
 */
API.Client.CodingList = function() {}

/**
 * @type {!API.Client.Metadata}
 * @export
 */
API.Client.CodingList.prototype.metadata;

/**
 * @type {!API.Client.Pagination}
 * @export
 */
API.Client.CodingList.prototype.pagination;

/**
 * @type {!Array<!API.Client.Coding>}
 * @export
 */
API.Client.CodingList.prototype.data;

